src/pip/_internal/commands/download.py

```diff
@@ -96,6 +97,7 @@ def run(self, options, args):
             options=options,
             session=session,
             target_python=target_python,
+            ignore_requires_python=options.ignore_requires_python,
```

Comment:
The `options` is already an argument, instead of adding a new argument `ignore_requires_python` - it can be accessed from `options` object inside the `Finder` factory method.